x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -115,22 +119,27 @@ export function getTemplate({
 export function generateMappings(fields: Field[]): IndexTemplateMappings {
   const dynamicTemplates: Array<Record<string, Properties>> = [];
   const dynamicTemplateNames = new Set<string>();
+  const runtimeFields: RuntimeFields = {};
 
   const { properties } = _generateMappings(fields, {
     addDynamicMapping: (dynamicMapping: {
       path: string;
       matchingType: string;
       pathMatch: string;
       properties: string;
+      runtimeProperties?: string;
     }) => {
       const name = dynamicMapping.path;
       if (dynamicTemplateNames.has(name)) {
         return;
       }
 
-      const dynamicTemplate: Properties = {
-        mapping: dynamicMapping.properties,
-      };
+      const dynamicTemplate: Properties = {};
+      if (dynamicMapping.runtimeProperties !== undefined) {
+        dynamicTemplate.runtime = dynamicMapping.runtimeProperties;
+      } else {
+        dynamicTemplate.mapping = dynamicMapping.properties;
+      }
```

Comment:
In a dynamic template, `mapping` and `runtime` cannot be defined at the same time in the object.